includes/WcfceMailProcessor.inc

```diff
@@ -189,10 +193,201 @@ class WcfceMailProcessor {
     $eml .= 'To: ' . implode($crlf . 'To: ', $recipients) . $crlf;
     $eml .= 'Subject: ' . $subject . $crlf . $crlf;
     $eml .= $body;
+    $eml = $this->appendWebformFilesToEml($eml, $message, $crlf);
+
+    return $eml;
+  }
+
+  /**
+   * Appends webform files as attachments to email text.
+   *
+   * @param string $eml
+   *   Email message in eml format.
+   * @param array|null $message
+   *   (Optional) Email data. If set to NULL then $this->message would be used.
+   * @param string $crlf
+   *   (Optional) End of line character.
+   *
+   * @return string
+   *   Email text (eml) with attachments.
+   */
+  public function appendWebformFilesToEml($eml, $message = NULL, $crlf = MAIL_LINE_ENDINGS) {
+    $attachments = $this->getWebformFilesAsEml($eml, $message, $crlf);
+
+    if (empty($attachments)) {
+      return $eml;
+    }
+
+    $boundary = $this->getEmailBoundary($eml);
+    // Eml final boundary looks like '--[boundary]--', but since we are adding
+    // attachments this boundary is not final now, hence we need to remove
+    // '--' at the end of the text (it would be added at the end of the files).
+    $eml = rtrim(preg_replace('!--' . $boundary . '--!', '--' . $boundary, $eml))
+      . $crlf
+      . rtrim($attachments) . '--';
 
     return $eml;
   }
 
+  /**
+   * Attaches files attached to a webform submission to activity.
+   *
+   * @param int $activity_id
+   *   Activity id.
+   *
+   * @return array|false
+   *   Entity data of the last file added, empty array or FALSE
+   *   in case of error.
+   *
+   * @throws \CiviCRM_API3_Exception
+   *   An exception generated by civicrm_api3().
+   */
+  public function attachWebformFilesToActivity($activity_id) {
+    if (!$activity_id) {
+      return FALSE;
+    }
+
+    $file_ids = $this->getWebformFileIds($this->message);
+
+    if (empty($file_ids)) {
+      return FALSE;
+    }
+
+    foreach ($file_ids as $file_id) {
+      $file = $this->getWebformFile($file_id);
+      $result = civicrm_api3('Attachment', 'create', [
+        'sequential' => 1,
+        'entity_table' => 'civicrm_activity',
+        'entity_id' => $activity_id,
+        'mime_type' => $file->filemime,
+        'name' => $file->filename,
+        'content' => $this->getWebformFile($file_id, 'content'),
+      ]);
+    }
+
+    return $result['values'][0] ?? [];
+  }
+
+  /**
+   * Encodes attached file contents to append them to email.
+   *
+   * @param string $headers
+   *   Email headers (used to get email boundary).
+   * @param array|null $message
+   *   (Optional) Email data. If set to NULL then $this->message would be used.
+   * @param string $crlf
+   *   (Optional) End of line character.
+   *
+   * @return string
+   *   Attached file contents (encoded), or empty string if no files found.
+   */
+  public function getWebformFilesAsEml($headers, $message = NULL, $crlf = MAIL_LINE_ENDINGS) {
```

Comment:
I think this function and most of the function in this class apart from `createActivity` should be private since these methods are not used outside this class.

Reply:
@ivan-compucorp The point here is to make the methods that will not be useful outside the class to be private by default. These methods I asked to make private are not used anywhere outside the class and there are no child class extending this class for now so it makes sense to keep them private as they are more or less useful for the class internals (For now).

When the need arises we can then make the function protected/public as the need arises.

Take for example the function `getEmailBoundary`, `appendWebformFilesToEml`, these methods are hardly useful outside the class. They accept parameters that is specific to what the class expects .
Another reason is that making a function public means you have to add unit tests for those functions in reality and that affects the way the function is written.  `appendWebformFilesToEml` for example accepts a parameter called `$eml`, for this function to be public means you need to add a test (in reality) and also add a validator that the correct EML format/string is passed from the consumer class/function. If this function is private, then you won't have to worry about this as the function is used internally by the class.

So you see deciding to make a function private/public is responsibility.

Reply:
Okay, since there is no big difference now, will change them to private.